### PR TITLE
[Bug] Consider backend status when more than one backends exists in same host 

### LINKED ISSUE #7783: [Bug] Consider backend status when more than one backends exists in same host
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

I have more than one BEs exist in same host. If one of them are dead, stream load process would fail sometimes. Because FE select one at random not alive one. This will cause 'No backend alive.' ERROR during stream load process.

### What You Expected?

Choose alive BE when more than one BE exists in same host, so that stream load process will not fail when there are both alive and dead BE in the same host.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengshengjun**
# Proposed changes

Issue Number: close #7783

## Problem Summary:

Consider backend status when more than one backends exists in same host 

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Comment by EmmyMiao87** [comment]
Please rebase your code to fix conflict ~

**Comment by morningman** [comment]
hi @zhengshengjun thanks for your PR. Please rebase to solve the conflict

**Comment by zhengshengjun** [comment]
> hi @zhengshengjun thanks for your PR. Please rebase to solve the conflict

conflict resolved ~

**Comment by zhengshengjun** [comment]
> Please rebase your code to fix conflict ~

Conflict resolved, could you please help me to check again ~

File: fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java [hunk 1]
```diff
@@ -871,6 +871,9 @@ public void releaseBackends(String clusterName, boolean isReplay) {
         List<Backend> backends = Lists.newArrayList();
         for (List<Backend> list : backendMaps.values()) {
             if (FeConstants.runningUnitTest || Config.allow_replica_on_same_host) {
+                if (needAlive) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java:874]: This block:
```
if (needAlive) {
    list = list.stream().filter(Backend::isAlive).collect(Collectors.toList());
}
```

should be added to the following `else` clause:

```
} else {
    if (needAlive) {
        list = list.stream().filter(Backend::isAlive).collect(Collectors.toList());
    }
    Collections.shuffle(list);
    backends.add(list.get(0));
}
```

Comment (zhengshengjun) [reply fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java]: Sorry for not having noticed the code have already changed too much, i have updated and tested.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**